docs/src/main/sphinx/connector/atop.rst

```diff
@@ -0,0 +1,66 @@
+==============
+Atop connector
+==============
+
+Overview
+--------
+
+The Atop connector supports reading disk utilization statistics from the ``atop``
+(Advanced System & Process Monitor) Linux server performance analysis tool.
+
+In order to use this connector, the machine on which the Trino worker is running
+needs to have the ``atop`` tool installed locally.
+
+Connector configuration
+-----------------------
+
+Create ``etc/catalog/atop.properties``
+
+.. code-block:: text
+
+    connector.name=atop
+    atop.executable-path=xxx
+
+Configuration variables
+-----------------------
+
+================================================ ====================== ========== =====================================================================================================
+Property Name                                    Default Value          Required   Description
+================================================ ====================== ========== =====================================================================================================
+``atop.concurrent-readers-per-node``             ``1``                  Yes        The number of concurrent read operations allowed per node.
+``atop.executable-path``                         (none)                 Yes        The file path on the local file system for the ``atop`` utility
+``atop.executable-read-timeout``                 ``1ms``                Yes        The timeout when reading from the atop process.
+``atop.max-history-days``                        ``30``                 Yes        The maximum number of days behind to take into account statistics.
+``atop.security``                                ``ALLOW_ALL``          Yes        The authorization rules for the connector. If set to ``FILE``, see :doc:`file-based rules </security/file-system-access-control>`.
+``atop.time-zone``                               System default         Yes        The timezone ID in which the atop data was collected. Generally the timezone of the host.
+================================================ ====================== ========== =====================================================================================================
+
+Usage
+-----
+
+The Atop connector provides a ``default`` schema.
+
+The tables available exposed by this connector can be retrieved by running ``SHOW TABLES``::
+
+    SHOW TABLES in atop.default;
+
+.. code-block:: text
+
+      Table
+    ---------
+     disks
+     reboots
+    (2 rows)
+
+
+The ``disks`` table offers disk utilization statistics recorded on the Trino node.
```

Comment:
The description of the tables is pretty shallow at the moment.
I don't know whether it makes sense to go more into details.

Reply:
If the metadata is always the same, it would be nice to have a list of columns with some description, if the name is not totally obvious. Something similar to https://trino.io/docs/current/connector/system.html#metadata-materialized-views

Reply:
Does it return a (set of) row(s) for every node? If yes, that would be worth mentioning.

Reply:
Shallow is much better than nothing .. thats what we got now. Also we can assume the user knows about the atop command or is willing to learn

Reply:
> Does it return a (set of) row(s) for every node? If yes, that would be worth mentioning.

Yes, it does.

```
trino> select * from atop.default.disks
    -> ;
  host_ip   |              start_time               |               end_time                | device_name | utilization_percent
------------+---------------------------------------+---------------------------------------+-------------+--------------------
 172.24.0.4 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:21:57.000 Europe/Vienna | nbd1        |                 0.0
 172.24.0.4 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:21:57.000 Europe/Vienna | nbd2        |                 0.0
 172.24.0.4 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:21:57.000 Europe/Vienna | nbd3        |                 0.0
 172.24.0.4 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:21:57.000 Europe/Vienna | nbd4        |                 0.0
 172.24.0.4 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:21:57.000 Europe/Vienna | nbd5        |                 0.0
 172.24.0.4 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:21:57.000 Europe/Vienna | nbd6        |                 0.0
 172.24.0.4 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:21:57.000 Europe/Vienna | nbd7        |                 0.0
 172.24.0.4 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:21:57.000 Europe/Vienna | nbd8        |                 0.0
 172.24.0.4 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:21:57.000 Europe/Vienna | nbd9        |                 0.0
 172.24.0.4 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:21:57.000 Europe/Vienna | nbd10       |                 0.0
 172.24.0.4 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:21:57.000 Europe/Vienna | nbd11       |                 0.0
 172.24.0.4 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:21:57.000 Europe/Vienna | nbd12       |                 0.0
 172.24.0.4 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:21:57.000 Europe/Vienna | nbd13       |                 0.0
 172.24.0.4 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:21:57.000 Europe/Vienna | nbd14       |                 0.0
 172.24.0.4 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:21:57.000 Europe/Vienna | nbd15       |                 0.0
 172.24.0.4 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:21:57.000 Europe/Vienna | vda         |                 0.0
 172.24.0.3 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:21:32.000 Europe/Vienna | nbd1        |                 0.0
 172.24.0.3 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:21:32.000 Europe/Vienna | nbd2        |                 0.0
 172.24.0.3 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:21:32.000 Europe/Vienna | nbd3        |                 0.0
 172.24.0.3 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:21:32.000 Europe/Vienna | nbd4        |                 0.0
 172.24.0.3 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:21:32.000 Europe/Vienna | nbd5        |                 0.0
 172.24.0.3 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:21:32.000 Europe/Vienna | nbd6        |                 0.0
 172.24.0.3 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:21:32.000 Europe/Vienna | nbd7        |                 0.0
 172.24.0.3 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:21:32.000 Europe/Vienna | nbd8        |                 0.0
 172.24.0.3 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:21:32.000 Europe/Vienna | nbd9        |                 0.0
 172.24.0.3 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:21:32.000 Europe/Vienna | nbd10       |                 0.0
 172.24.0.3 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:21:32.000 Europe/Vienna | nbd11       |                 0.0
 172.24.0.3 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:21:32.000 Europe/Vienna | nbd12       |                 0.0
 172.24.0.3 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:21:32.000 Europe/Vienna | nbd13       |                 0.0
 172.24.0.3 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:21:32.000 Europe/Vienna | nbd14       |                 0.0
 172.24.0.3 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:21:32.000 Europe/Vienna | nbd15       |                 0.0
 172.24.0.3 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:21:32.000 Europe/Vienna | vda         |                 0.0
 172.24.0.5 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:22:26.000 Europe/Vienna | nbd1        |                 0.0
 172.24.0.5 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:22:26.000 Europe/Vienna | nbd2        |                 0.0
 172.24.0.5 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:22:26.000 Europe/Vienna | nbd3        |                 0.0
 172.24.0.5 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:22:26.000 Europe/Vienna | nbd4        |                 0.0
 172.24.0.5 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:22:26.000 Europe/Vienna | nbd5        |                 0.0
 172.24.0.5 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:22:26.000 Europe/Vienna | nbd6        |                 0.0
 172.24.0.5 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:22:26.000 Europe/Vienna | nbd7        |                 0.0
 172.24.0.5 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:22:26.000 Europe/Vienna | nbd8        |                 0.0
 172.24.0.5 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:22:26.000 Europe/Vienna | nbd9        |                 0.0
 172.24.0.5 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:22:26.000 Europe/Vienna | nbd10       |                 0.0
 172.24.0.5 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:22:26.000 Europe/Vienna | nbd11       |                 0.0
 172.24.0.5 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:22:26.000 Europe/Vienna | nbd12       |                 0.0
 172.24.0.5 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:22:26.000 Europe/Vienna | nbd13       |                 0.0
 172.24.0.5 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:22:26.000 Europe/Vienna | nbd14       |                 0.0
 172.24.0.5 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:22:26.000 Europe/Vienna | nbd15       |                 0.0
 172.24.0.5 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:22:26.000 Europe/Vienna | vda         |                 0.0
 172.24.0.5 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:24:25.000 Europe/Vienna | nbd1        |                 0.0
 172.24.0.5 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:24:25.000 Europe/Vienna | nbd2        |                 0.0
 172.24.0.5 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:24:25.000 Europe/Vienna | nbd3        |                 0.0
 172.24.0.5 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:24:25.000 Europe/Vienna | nbd4        |                 0.0
 172.24.0.5 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:24:25.000 Europe/Vienna | nbd5        |                 0.0
 172.24.0.5 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:24:25.000 Europe/Vienna | nbd6        |                 0.0
 172.24.0.5 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:24:25.000 Europe/Vienna | nbd7        |                 0.0
 172.24.0.5 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:24:25.000 Europe/Vienna | nbd8        |                 0.0
 172.24.0.5 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:24:25.000 Europe/Vienna | nbd9        |                 0.0
 172.24.0.5 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:24:25.000 Europe/Vienna | nbd10       |                 0.0
 172.24.0.5 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:24:25.000 Europe/Vienna | nbd11       |                 0.0
 172.24.0.5 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:24:25.000 Europe/Vienna | nbd12       |                 0.0
 172.24.0.5 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:24:25.000 Europe/Vienna | nbd13       |                 0.0
 172.24.0.5 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:24:25.000 Europe/Vienna | nbd14       |                 0.0
 172.24.0.5 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:24:25.000 Europe/Vienna | nbd15       |                 0.0
 172.24.0.5 | 2022-01-18 04:37:05.000 Europe/Vienna | 2022-01-18 10:24:25.000 Europe/Vienna | vda         |                 0.0

Query 20220118_092454_00007_fsins, FINISHED, 3 nodes
Splits: 11 total, 11 done (100.00%)
0.23 [64 rows, 0B] [274 rows/s, 0B/s]
```

Reply:
This is out of the scope of this PR, but it's a bit unfortunate there's only a `host_ip` column and not a node id one. I wonder which IP is it if there are multiple interfaces on a host.

Thanks for adding the description!

Reply:
The ip is the Trino worker IP.

```
select * from system.runtime.nodes;
```
See https://github.com/trinodb/trino/blob/master/plugin/trino-atop/src/main/java/io/trino/plugin/atop/AtopSplitManager.java#L83